drift/core/psestimation.py

```diff
@@ -625,16 +627,55 @@ def q_estimator(self, mi, vec1, vec2=None, noise=False):
         x1 = np.dot(evecs.T.conj(), x0)
 
         # Project back into sky basis
-        x2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(mi, x1, conj=True)
+        x2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(
+            mi, x1, temponly=True, conj=True
+        )
+        # Slice array for temponly
+        x2 = x2[:, 0]
 
         if vec2 is not None:
             y0 = (vec2.T / (evals + 1.0)).T
             y1 = np.dot(evecs.T.conj(), x0)
-            y2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(mi, x1, conj=True)
+            y2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(
+                mi, x1, temponly=True, conj=True
+            )
+            # Slice array for temponly
+            y2 = y2[:, 0]
+
         else:
             y0 = x0
             y2 = x2
 
+        return x2, y2
+
+    def q_estimator(self, mi, vec1, vec2=None, noise=False):
+        """Estimate the q-parameters from given data (see paper).
+
+        Parameters
+        ----------
+        mi : integer
+            The m-mode we are calculating for.
+        vec : np.ndarray[num_kl, num_realisatons]
+            The vector(s) of data we are estimating from. These are KL-mode
+            coefficients.
+        noise : boolean, optional
+            Whether we should project against the noise matrix. Used for
+            estimating the bias by Monte-Carlo. Default is False.
+
+        Returns
+        -------
+        qa : np.ndarray[numbands]
+            Array of q-parameters. If noise=True then the array is one longer,
+            and the last parameter is the projection against the noise.
+        """
+
+        evals, evecs = self.kltrans.modes_m(mi)
+
+        x2, y2 = self.project_vector_kl_to_sky(mi, vec1, vec2=None)
```

Comment:
Last argument should be `vec2=vec2` instead of `vec2=None`, no?

Reply:
vec2 = None is default. Only pass vec2 if you want think you want to compute a cross power spectrum.